packages/eslint-plugin/src/rules/no-unused-expressions.ts

```diff
@@ -19,6 +19,7 @@ export default createRule<Options, MessageIds>({
     docs: {
       description: 'Disallow unused expressions',
       extendsBaseRule: true,
+      recommended: 'stylistic',
```

Comment:
Honestly this should be strict and could even be recommended. There's no reason to have unused expression cruft ever.